### DISCUSSION TITLE
Font merging improvement

### DISCUSSION THREAD

**Discussion by sammyfreg** [ocornut/imgui#5014] [New Users Build/Link/Run issues ONLY!]
I am currently adding support of Japanese font to [UnrealNetImgui](https://github.com/sammyfreg/UnrealNetImgui) and I figured, might as well add some icons support, after reading the font [tutorial](https://github.com/ocornut/imgui/blob/master/docs/FONTS.md).

It is working well (tutorials helps a lot, thanks) but I noticed a potential memory improvement that could be nice. I was thinking of adding multiple fonts with multiple sizes and append the icons glyphs to each. 

**--- Example ---**
**Font 1:** Cousine Fixed (16px) + FontAwesome (16px)
**Font 2:** Cousine Fixed (24px) + FontAwesome (24px)
**Font 3:** Droid Sans (16px) + FontAwesome (16px)
**Font 4:** Droid Sans (24px) + FontAwesome (24px)
**Font n:** ...

Looking at the font inspector, I can see that even though the icon glyphs might be identical, they are allocated a new space in texture memory per font entry. My current font setup requires a 4kx8k texture! :D  

When building the Font Atlas, it would be nice to detect if a font with same size already has been added, and just re-use its texture entry. That way, all `FontAwesome 16px` could share the same texture area instead of allocating more.

Same would be true, if wanting each roman font to also have Japanese support, without duplicating the glyphs.

**Comment by ocornut** [comment]
That would be great indeed. Feel like working on a PR ? :)

**Reply by sammyfreg** [reply]
Seems like I might at some point :)
